src/Logging/PsrLogger.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Logging;
+
+use Psr\Log\LoggerInterface;
+
+class PsrLogger implements LoggerInterface
```

Comment:
Great points!

What if the `resource` on an entry became optional instead of required, giving you the ability to optionally override the resource that is configured by default?